scipy/stats/_continuous_distns.py

```diff
@@ -9244,6 +9244,35 @@ def _entropy(self, kappa):
         return (-kappa * sc.i1e(kappa) / sc.i0e(kappa) +
                 np.log(2 * np.pi * sc.i0e(kappa)) + kappa)
 
+    @extend_notes_in_docstring(rv_continuous, notes="""\
+        The limits of integration of the vonmises distribution
+        default to the 2pi interval centered at loc
+        (e.g. [-pi,pi] when ``loc=0``).
+
+        >>> r = vonmises(kappa=1).expect(lambda x: np.exp(1j*x))
+        >>> np.angle(r)
+        0.0
+
+        The ``vonmises::expect`` method, as shown above, uses the
+        ``complex_func`` feature of ``integrate.quad`` to evaluate
+        integral.\n\n""")
+    def expect(self, func=None, args=(), loc=0, scale=1, lb=-np.pi, ub=np.pi,
+               conditional=False, **kwds):
+        self._argcheck(*args)
+        _a, _b = -np.pi, np.pi
+        if func is None:
+            def func(x, *args):
+                return np.exp(1j*x)
```

Comment:
What you have now would make a lot of sense when we actually have a complex distribution infrastructure.  But for now, let's keep it backward-compatible.
```suggestion
```